Packs/MitreCaldera/Integrations/MitreCaldera/MitreCaldera.py

```diff
@@ -1323,6 +1325,133 @@ def get_objectives_command(client: Client, args: Dict[str, Any]) -> CommandResul
+
+    Returns:
+        list: A list of dictionaries containing only the operations where the client's name is present in the "name" field.
+    """
+    return [operation for operation in operations if client_name in operation.get("name")]
+
+
+def operation_to_incident(operation: dict, operation_date: str) -> dict:
+    """
+    Converts an operation dictionary into an incident dictionary.
+
+    Args:
+        operation (dict): A dictionary containing details of the operation.
+        operation_date (str): The date when the operation occurred.
+
+    Returns:
+        dict: A dictionary representing the incident, including the operation's ID, name, and the date it occurred.
+    """
+    operation_id = operation.get("id")
+    operation_name = operation.get("name")
+    incident = {
+        "name": f"Caldera: {operation_id} {operation_name}",
+        "occured": f"{operation_date}",
+        "rawJSON": json.dumps(operation),
+    }
+    return incident
+
+
+def operations_to_incidents(operations: list, last_fetch_datetime: str) -> tuple[list, str]:
+    """
+    Converts a list of operations into a list of incidents and updates the latest incident time.
+
+    Args:
+        operations (list): A list of dictionaries, where each dictionary represents an operation.
+        last_fetch_datetime (str): The datetime string representing the last time incidents were fetched.
+
+    Returns:
+        tuple: A tuple containing:
+            - A list of dictionaries, each representing an incident.
+            - A string representing the latest incident time.
+    """
+    incidents: List[Dict[str, str]] = []
+    latest_incident_time = last_fetch_datetime
+
+    for operation in operations:
+        operation_datetime = operation.get("start", "")
+        incident = operation_to_incident(operation, operation_datetime)
+        incidents.append(incident)
+
+        if datetime.strptime(normalize_date_str(operation_datetime), DATE_FORMAT) > datetime.strptime(
+            normalize_date_str(latest_incident_time), DATE_FORMAT
+        ):
+            latest_incident_time = operation_datetime
+
+    return incidents, latest_incident_time
+
+
+def fetch_incidents(client: Client, params: Dict[str, str]):
+    last_run: Dict[str, str] = demisto.getLastRun()
+    demisto.debug(f"[Caldera] last run: {last_run}")
+
+    last_fetch = get_last_fetch_time(last_run, params)
+    demisto.debug(f"[Caldera] last fetch is: {last_fetch}")
+
+    operations = client.get_operations(None, [], [])
+
+    # Fetch only operations after last fetch time
+    operations = [
+        operation for operation in operations if datetime.strptime(
+            normalize_date_str(operation.get("start")), DATE_FORMAT) > datetime.strptime(
+            normalize_date_str(last_fetch), DATE_FORMAT)
+    ]
+
+    if client_name := params.get("client_name"):
+        operations = filter_operations(operations, client_name)
+
```

Comment:
consider to add this condition inside **filter_operations** method:

` if client_name := params.get("client_name"):`

and also the last fetch time and than call **filter_operations** like this:
```suggestion

        operations = filter_operations(operations, client_name, last_time)

```